braindecode/models/eegnet.py

```diff
@@ -265,26 +264,35 @@ def __init__(
 
         # Incorporating classification module and subsequent ones in one final layer
         module = nn.Sequential()
+        if not final_layer_linear:
+            module.add_module(
+                "conv_classifier",
+                nn.Conv2d(
+                    self.F2,
+                    self.n_outputs,
+                    (n_out_virtual_chans, self.final_conv_length),
+                    bias=True,
+                ),
+            )
 
-        module.add_module(
-            "conv_classifier",
-            nn.Conv2d(
-                self.F2,
-                self.n_outputs,
-                (n_out_virtual_chans, self.final_conv_length),
-                bias=True,
-            ),
-        )
-
-        # Transpose back to the logic of braindecode,
-        # so time in third dimension (axis=2)
-        module.add_module(
-            "permute_back",
-            Rearrange("batch x y z -> batch x z y"),
-        )
-
-        module.add_module("squeeze", Expression(squeeze_final_output))
+            # Transpose back to the logic of braindecode,
+            # so time in third dimension (axis=2)
+            module.add_module(
+                "permute_back",
+                Rearrange("batch x y z -> batch x z y"),
+            )
 
+            module.add_module("squeeze", Expression(squeeze_final_output))
+        else:
+            module.add_module("flatten", nn.Flatten())
+            module.add_module(
+                "linearconstraint",
+                LinearWithConstraint(
+                    in_features=output_shape[1] * output_shape[3],
```

Comment:
```suggestion
                    in_features=self.F2 * self.final_conv_length,
```
(For consistency with the convolutional layer version)

Plus, in case `n_out_virtual_chans!=1`, it doesn't work. Is it possible to be in this case?
- If yes, we need `in_features=self.F2 * n_out_virtual_chans  * self.final_conv_length`
- Otherwise, we should replace `n_out_virtual_chans` everywere to simplify the code and put an `assert output_shape[2]==1`